app/assets/javascripts/app/views/publisher/mention_view.js

```diff
@@ -0,0 +1,224 @@
+    "blur #status_message_fake_text": "onInputBoxBlur",
+  },
+
+  initialize: function(){
+    this.mentionedPeople = [];
+
+    // contains the 'fake text' displayed to the user
+    // also has a data-messageText attribute with the original text
+    this.inputBox = this.$("#status_message_fake_text");
+    // contains the mentions displayed to the user
+    this.mentionsBox = this.$(".mentions-box");
+    this.typeaheadInput = this.$(".typeahead-mention-box");
+    this.bindTypeaheadEvents();
+
+    app.views.SearchBase.prototype.initialize.call(this, {
+      typeaheadInput: this.typeaheadInput,
+      customSearch: true,
+      autoselect: true
+    });
+  },
+
+  bindTypeaheadEvents: function(){
+    var self = this;
+    // Process mention when the user selects a result.
+    this.typeaheadInput.on("typeahead:select", function(evt, person){ self.onSuggestionSelection(person); });
+  },
+
+  addPersonToMentions: function(person){
+    if(!person || !person.name || !person.handle){ return; }
+    // This is needed for processing preview
+    /* jshint camelcase: false */
+    person.diaspora_id = person.handle;
+    /* jshint camelcase: true */
+    this.mentionedPeople.push(person);
+    this.ignorePersonForSuggestions(person);
+  },
+
+  cleanMentionedPeople: function(){
+    var inputText = this.inputBox.val();
+    this.mentionedPeople = _.reject(this.mentionedPeople, function(person){
+      return !person.name || inputText.indexOf(person.name) === -1;
+    });
+    this.ignoreDiasporaIds = _.map(this.mentionedPeople, function(person){ return person.handle; });
+  },
+
+  onSuggestionSelection: function(person){
+    var messageText = this.inputBox.val();
+    var caretPosition = this.inputBox[0].selectionStart;
+    var triggerCharPosition = messageText.lastIndexOf(this.triggerChar, caretPosition);
+
+    if(triggerCharPosition === -1) { return; }
+
+    this.addPersonToMentions(person);
+    this.closeSuggestions();
+
+    messageText = messageText.substring(0, triggerCharPosition) +
+      this.invisibleChar + person.name + messageText.substring(caretPosition);
+
+    this.inputBox.val(messageText);
+    this.updateMessageTexts();
+
+    this.inputBox.focus();
+    var newCaretPosition = triggerCharPosition + person.name.length + 1;
+    this.inputBox[0].setSelectionRange(newCaretPosition, newCaretPosition);
+  },
+
+  /**
+   * Replaces every combination of this.invisibleChar + mention.name by the
+   * correct syntax for both hidden text and visible one.
+   *
+   * For instance, the text "Hello \u200Buser1" will be tranformed to
+   * "Hello @{user1 ; user1@pod.tld}" in the hidden element and
+   * "Hello <strong><span>user1</span></strong>" in the element visible to the user.
+   */
+  updateMessageTexts: function(){
+    var fakeMessageText = this.inputBox.val(),
+        mentionBoxText = fakeMessageText,
+        messageText = fakeMessageText;
+
+    var self = this;
+    _.each(this.mentionedPeople, function(person){
```

Comment:
I'm wondering if we should start using `Array.prototype.forEach()` which is supported everywhere but IE 8 and older. Reducing the overall underscore usage where it is not necessary could be a good thing. Opinions?